lib/UserBackend.php

```diff
@@ -81,6 +83,28 @@ public function __construct(
 		$this->userData = $userData;
 	}
 
+    /**
+     * checks whether the user is allowed to change his avatar in Nextcloud
+     *
+     * @param string $uid the Nextcloud user name
+     * @return boolean either the user can or cannot
+     * @throws \Exception
+     */
+    public function canChangeAvatar($uid) {
+        if (!$this->implementsActions(Backend::PROVIDE_AVATAR)) {
+            return true;
+        }
+        try {
+            if (empty(trim($this->getAttributeKeys('saml-attribute-mapping-avatar_mapping')[0]))) {
+                return true;
+            }
+        } catch(\InvalidArgumentException $e) {
+            return true;
```

Comment:
:thinking: false?

Also, are you sure you don't want to check the return of `getAttributeValue` instead a few lines above? You can save the if and return the result of the comparison.

Reply:
Yes,I refactor that if statement! The method evaluates if a user can change their avatar manually via the Nextcloud UI. If the argument is not existent in the user_saml configuration it is the same as being empty and the user can change their avatar. So the catch should be fine I think.